apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnection.cs

```diff
@@ -428,7 +428,8 @@ public override Task OpenAsync(CancellationToken cancellationToken)
                 {
                     if (string.IsNullOrEmpty(_connectionStringBuilder?.SpannerDatabase))
                     {
-                        Logger.Warn(() => "No database was defined. Therefore OpenAsync did not establish a session.");
+                        Logger.Info(() => "No database was defined. Therefore OpenAsync did not establish a session.");
+                        _state = ConnectionState.Open;
```

Comment:
It's not clear why the state would be open if we don't have a database. What can be done with this connection afterwards? Can it be treated like other open connections? Should we throw instead?

Reply:
The state needs to be considered open because some of EF will not allow Ddl statements unless the connection is opened. 
I purposely allow specifying a connection string without a database (but with project+instance) because spanner has no "master" db concept.  But it was an oversight to not set the state to opened when that happened.